services/data/db_utils.py

```diff
@@ -66,29 +69,12 @@ def get_latest_attempt_id_for_tasks(artifacts):
     return attempt_ids
 
 
-def filter_artifacts_for_latest_attempt(artifacts):
-    """
-    filters artifacts of latest task attempt. 
-    Args:
-        artifacts (List[ArtifactRow]): a dictionary form of `ArtifactRow`
-
-    Returns:
-        `list` of filtered artifacts for the latest attempt
-    """
+def filter_artifacts_for_latest_attempt(artifacts:List[ArtifactRow]) -> List[ArtifactRow]:
```

Comment:
🙈 ; I have changed to type hints but they are not the most accurate. This change has some nuance; `List[ArtifactRow]` is, in fact, list having a dictionary representation `ArtifactRow`